templates/profile/kubernetes/haproxy/haproxy.cfg.erb

```diff
@@ -1,31 +1,49 @@
 # Managed by puppet (nebula/profile/kubernetes/haproxy/haproxy.cfg.erb)
 
 global
+  # Debian 11 defaults
   log /dev/log  local0
   log /dev/log  local1 notice
   chroot /var/lib/haproxy
-  stats socket /run/haproxy/admin.sock mode 660 level admin
+  stats socket /run/haproxy/admin.sock mode 660 level admin expose-fd listeners
   stats timeout 30s
   user haproxy
   group haproxy
   daemon
   ca-base /etc/ssl/certs
   crt-base /etc/ssl/private
+  # 2018-08-10; tuning for our level of usage
   maxconn 16384
 
-ssl-default-bind-ciphers ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-CCM:DHE-RSA-AES256-CCM8:DHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-CCM:DHE-RSA-AES128-CCM8:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES256-SHA:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA
+  # adapted from https://ssl-config.mozilla.org/#server=haproxy
+  ssl-default-bind-curves X25519:prime256v1:secp384r1
+  ssl-default-bind-ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384
+  ssl-default-bind-ciphersuites TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
+  ssl-default-bind-options prefer-client-ciphers ssl-min-ver TLSv1.2 no-tls-tickets
 
-  ssl-default-bind-options no-sslv3
-  tune.ssl.default-dh-param 2048
+  ssl-default-server-curves X25519:prime256v1:secp384r1
+  ssl-default-server-ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384
+  ssl-default-server-ciphersuites TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
+  ssl-default-server-options ssl-min-ver TLSv1.2 no-tls-tickets
 
 defaults
+  # Debian 11 defaults
   log  global
   mode  http
   option  httplog
   option  dontlognull
   timeout connect 5000
+  # 2018-08-10; prevent observed 504 errors
   timeout client  180000
   timeout server  360000
+
```

Comment:
I added this useless whitespace to make the files align. It would be better if we could add the `errorfile` directives and make them match. Then we could even use the one template in both places if we want. Thoughts?